openmc/tallies.py

```diff
@@ -1425,25 +1436,47 @@ def export_results(self, filename='tally-results', directory='.',
             # Pickle the Tally results to a file
             pickle.dump(tally_results, open(filename, 'wb'))
 
-    def _outer_product(self, other, binary_op):
+    def hybrid_product(self, other, binary_op, filter_product=None,
+                        nuclide_product=None, score_product=None):
         """Combines filters, scores and nuclides with another tally.
 
-        This is a helper method for the tally arithmetic methods. The filters,
-        scores and nuclides from both tallies are enumerated into all possible
-        combinations and expressed as CrossFilter, CrossScore and
-        CrossNuclide objects in the new derived tally.
+        This is a helper method for the tally arithmetic methods. It is called a
+        "hybrid product" because it performs a combination of tensor
+        (or Kronecker) and entrywise (or Hadamard) products. The filters from
+        both tallies are combined using an entrywise (or Hadamard) product on
+        matching filters. By default, if all nuclides are identical in the two
+        tallies, the entrywise product is performed across nuclides; else the
+        tensor product is performed. By default, if all scores are identical in
+        the two tallies, the entrywise product is performed across scores; else
+        the tensor product is performed. Users can also call the method
+        explicitly and specify the desired product.
 
         Parameters
         ----------
         other : Tally
-            The tally on the right hand side of the outer product
+            The tally on the right hand side of the hybrid product
         binary_op : {'+', '-', '*', '/', '^'}
-            The binary operation in the outer product
+            The binary operation in the hybrid product
+        filter_product : str, optional
+            The type of product (tensor or entrywise) to be performed between
+            filter data. The default is the entrywise product. Currently only
+            the entrywise product is supported since a tally cannot contain
```

Comment:
I don't see any obvious use cases that would warrant someone to spend the time to implement this soon, so I would lean on the side of not creating an issue since it might be up there for a while. A general use case where a filter tensor product could be useful is getting ratios of tallies between different filters. For instance, getting the ratio of the fast-to-thermal absorption rate in a material could be achieved with only one tally by using an two-energy-group filter and performing a tensor product across filters instead of using two separate tallies.